### PR TITLE
Enable explicit namespaces

### PR DESCRIPTION
**Pull request by ferruzzi**
The recent PR to reorganize the package requires a change to explicit namespaces for at least some IDEs (PyCharm for sure, possibly others).

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/__init__.py [hunk 1]
```diff
@@ -17,6 +17,9 @@
 # under the License.
 from __future__ import annotations
 
+# Explicit namespaces are required by some IDEs to resolve the import paths.
+__path__ = __import__("pkgutil").extend_path(__path__, __name__)  # type: ignore
```
Comment (ashb) [inline airflow/__init__.py:23]: Delete ~L67 now

File: providers/src/airflow/providers/__init__.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from __future__ import annotations
+
+# Explicit namespaces are required by some IDEs to resolve the import paths.
```
Comment (ashb) [inline providers/src/airflow/providers/__init__.py]:
```suggestion
# Explicit namespaces are required by some IDEs to resolve the import paths.
#
# Note: this file is not installed or distributed in any distribution!
```

**Comment by gopidesupavan** [comment]
@ferruzzi Just wanted to let you know, the lowest direct dependency test failures are in microsoft provider, it is due to recent release of microsoft-kiota-http. imports are not working. am trying to fix these deps on different pr.

File: providers/src/airflow/providers/__init__.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+# under the License.
+
+from __future__ import annotations
+
+# Make `airflow` a namespace package, supporting installing
+# airflow.providers.* in different locations (i.e. one in site, and one in user
+# lib.)  This is required by some IDEs to resolve the import paths.
+#
+# Note: this file is not installed or distributed in any distribution!
+__path__ = __import__("pkgutil").extend_path(__path__, __name__)  # type: ignore
```
Comment (kaxil) [inline providers/src/airflow/providers/__init__.py:26]: I think this should be before any imports

Comment (ferruzzi) [reply providers/src/airflow/providers/__init__.py:26]: PyCharm's linter complains that `from __future__ imports must occur at the beginning of the file `.

Comment (ferruzzi) [reply providers/src/airflow/providers/__init__.py:26]: In fact, I just double checked and out static checks enforce that as well.  Going to resolve this.

**Comment by shahar1** [comment]
There's an error when using `breeze build-docs`:
```
/opt/airflow/docs/apache-airflow-providers-<PROVIDER_NAME>/_api/airflow/providers/index.rst:document isn't included in any
toctree
```

File: docs/conf.py [hunk 4]
```diff
@@ -765,7 +765,7 @@ def _get_params(root_schema: dict, prefix: str = "", default_section: str = "")
     # These sub-folders aren't really providers, but we need __init__.py files else various tools (ruff, mypy)
     # get confused by providers/tests/systems/cncf/kubernetes and think that folder is the top level
     # kubernetes module!
-    "*/providers/tests/__init__.py",
+    "*/providers/src/airflow/providers/__init__.py*/providers/tests/__init__.py",
```
Comment (shahar1) [inline docs/conf.py]:
```suggestion
    "*/providers/src/airflow/providers/__init__.py",
    "*/providers/tests/__init__.py",
```

File: airflow/__init__.py [hunk 5]
```diff
@@ -17,6 +17,11 @@
 # under the License.
 from __future__ import annotations
 
+# Make `airflow` a namespace package, supporting installing
+# airflow.providers.* in different locations (i.e. one in site, and one in user
+# lib.)  This is required by some IDEs to resolve the import paths.
+__path__ = __import__("pkgutil").extend_path(__path__, __name__)  # type: ignore
```
Comment (potiuk) [inline airflow/__init__.py:23]: Nice!

File: docs/conf.py (hunk 4 above)
Comment (potiuk) [reply docs/conf.py]: Yeah. Accidental change.

**Comment by potiuk** [comment]
You need to rebase @ferruzzi -> all the problems you see in here have been solved in main.

**Comment by ferruzzi** [comment]
Alright.  This is all green, up-to-date, and typo-free.

**Review by vincbeck — APPROVED** [review]

**Review by shahar1 — APPROVED** [review]

**Pull request merged**